### PR TITLE
Use MSVS pinned version for installs in Docker

### PR DESCRIPTION
**Pull request by arjungm**
b/316212112

Change-Id: I03c0ee23c8922d6a668a0f34f3ec7d3b317f5e8d

### CODE REVIEW

File: docker/windows/base/visualstudio2022/Dockerfile [hunk 1]
```diff
@@ -22,9 +22,10 @@ FROM ${FROM_IMAGE}
 SHELL ["powershell", "-ExecutionPolicy", "Unrestricted", "-Command", `
        "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
 
+ARG MSVS_INSTALLER_URL="https://download.visualstudio.microsoft.com/download/pr/63b5064f-af60-4cbe-96cd-a9dd9d41ee3d/436695f5b9083e37a62f63326dd4c2757b5780316f2c7e6ba0fdcee0770bfe67/vs_BuildTools.exe"
 RUN mkdir C:\TEMP;`
     Write-Host ('Downloading vs_buildtools.exe');`
-    Invoke-WebRequest -Uri https://aka.ms/vs/17/release/vs_buildtools.exe `
+    Invoke-WebRequest -Uri $MSVS_INSTALLER_URL `
```
Comment (kaidokert) [inline docker/windows/base/visualstudio2022/Dockerfile:28]: To be clear, doing this doesn't pin the versions of components the installer installs. It's safe to use the latest installer.

If your intent is to pin components, we have to pin and possibly vendor the manifest files themselves

Comment (niranjanyardi) [reply docker/windows/base/visualstudio2022/Dockerfile:28]: this should be able to pin the components installed : https://buganizer.corp.google.com/issues/316212112#comment4 . idk if arjun tried this locally using vs_professional.exe : if you install https://download.visualstudio.microsoft.com/download/pr/7aa16be3-9952-4bd2-8ecf-eae91faa0a06/ad45abc90e9574810fd9561e4b9a8bb3aff328384b43b7db6277102de4f2e3bf/vs_Professional.exe  it should have an earlier version of llvm -clang component inspite of using the same script: tools/install_visual_studio_22.ps1

Comment (arjungm) [reply docker/windows/base/visualstudio2022/Dockerfile:28]: good catch, I didn't change the helper install script

Comment (arjungm) [reply docker/windows/base/visualstudio2022/Dockerfile:28]: > doing this doesn't pin the versions of components the installer installs

if that is the case, then could you elaborate on what "vendor the manifest files" means?

Comment (kaidokert) [reply docker/windows/base/visualstudio2022/Dockerfile]: I pinged you 1:1 on this, and my response to the question is b/316212112#comment3

But it seems your implementation is actually correct and much better here - the "vendoring" of manifest files is much higher overhead than this code here. I'll resolve the comment and assume this path here works fine.

**Comment by niranjanyardi** [comment]
I verified that this approach works locally by downloading the professional version and getting clang 15 from it.

Note that tools/install_visual_studio_22.ps1 needs to be updated too.

Im adding a +2 proactively given you ll fix this in this change.

**Review by niranjanyardi — APPROVED** [review]

**Comment by isarkis** [comment]
Please make sure this change passes Docker image build on GCB, see http://go/lbreview/271701

**Pull request merged**